.github/workflows/pre-commit.yml

```
@@ -10,5 +10,5 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: actions/setup-python@v3
+    - uses: actions/setup-python@v4
```

```suggestion
    - uses: actions/setup-python@v4
```

This needs a SHA pin.
/hold